File: Rust/Arrays & Hashing/Maximum-Ice-Cream-Bars.rs
pub struct Solution;

impl Solution {
    pub fn max_ice_cream(costs: Vec<i32>, coins: i32) -> i32 {
        let mut costs: Vec<i32> = costs;
        costs.sort();
        let mut coins = coins;
        let mut round = 0;

        for cost in costs {
            if coins < cost {
                break;
            }
            round += 1;
            coins -= cost;
        }
        round
    }
}
fn main() {}

File: Javascript/Maximum-Ice-Cream-Bars.js
/**
 * @param {number[]} costs
 * @param {number} coins
 * @return {number}
 */
const maxIceCream = (costs, coins) => {
  costs.sort((a, b) => a - b);
  let round = 0;

  for (cost of costs) {
    if (coins < cost) {
      break;
    }

    coins -= cost;
    round++;
  }
  return round;
};
